pkg/libvirt/domain.go

```diff
@@ -177,16 +177,18 @@ func virtiofsSupported(conn *libvirt.Connect) error {
 		return fmt.Errorf("Error parsing libvirt domain capabilities: %w", err)
 	}
 
-	if caps.Devices.FileSystem.Supported != "yes" {
-		return drivers.ErrNotSupported
-	}
-	for _, enum := range caps.Devices.FileSystem.Enums {
-		if enum.Name != "driverType" {
-			continue
+	if caps.Devices.FileSystem != nil {
```

Comment:
I think an early return will be enough before we try to access `caps.Devices.FileSystem.Supported`, just to avoid the nested if, both achieves the same functionality
```go
if caps.Devices.FileSystem == nil {
   return drivers.ErrNotSupported
}
```

Reply:
yes right, updated